src/overlays/actors/ovl_Bg_Hakugin_Post/z_bg_hakugin_post.c

```diff
@@ -21,103 +54,994 @@ const ActorInit Bg_Hakugin_Post_InitVars = {
+                    (unkStruct->unk_0000[j].unk_14.y + D_80A9D880[unkStruct->unk_0000[j].unk_00].unk_00)) {
+                    unkStruct->unk_0000[i].unk_14.y =
+                        unkStruct->unk_0000[j].unk_14.y + D_80A9D880[unkStruct->unk_0000[j].unk_00].unk_00;
+                    unkStruct->unk_0000[i].unk_24 = 0.0f;
+                    if ((unkStruct->unk_0000[i].unk_2F & 1) != 0) {
+                        unkStruct->unk_0000[i].unk_2F &= ~2;
+                    } else {
+                        unkStruct->unk_0000[i].unk_2F |= 3;
+                    }
+                } else {
+                    unkStruct->unk_0000[i].unk_2F &= ~3;
+                }
+            }
+        }
+
+        if (unkStruct->unk_0000[i].unk_34 == 2) {
+            if (Math3D_XZLengthSquared(unkStruct->unk_0000[i].unk_14.x, unkStruct->unk_0000[i].unk_14.z) > 193600.02f) {
+                unkStruct->unk_0000[i].unk_34 = 3;
+            }
+        } else if (unkStruct->unk_0000[i].unk_34 == 3) {
+            if (Math3D_XZLengthSquared(unkStruct->unk_0000[i].unk_14.x, unkStruct->unk_0000[i].unk_14.z) > 278784.03f) {
+                func_80A9B554(this, globalCtx, unkStruct, &unkStruct->unk_0000[i]);
+                func_8019F128(NA_SE_EV_GLASSBROKEN_IMPACT);
+                unkStruct->unk_0000[i].unk_34 = 4;
+                unkStruct->unk_0000[i].unk_30 = 30;
+            }
+        } else if (unkStruct->unk_0000[i].unk_34 == 4) {
+            unkStruct->unk_0000[i].unk_30--;
+            if (unkStruct->unk_0000[i].unk_30 <= 0) {
+                unkStruct->unk_0000[i].unk_34 = 5;
+            }
+        }
+    }
+}
+
+void func_80A9C058(BgHakuginPost* this, GlobalContext* globalCtx, BgHakuginPostUnkStruct* unkStruct) {
+    s32 i;
+    s32 pad;
+    Vec3f sp44;
+    s32 pad2[2];
+    s16 sp3A;
+
+    if (this->unk_170 == 0) {
+        for (i = 0; i < unkStruct->count; i++) {
+            BgHakuginPostUnkStruct1* unkStruct1 = &unkStruct->unk_0000[i];
+
+            if (unkStruct1->unk_2F & 2) {
+                sp44.x = this->dyna.actor.home.pos.x + unkStruct1->unk_14.x;
+                sp44.y = this->unk_16C + unkStruct1->unk_14.y;
+                sp44.z = this->dyna.actor.home.pos.z + unkStruct1->unk_14.z;
+                func_8013ECE0(Math3D_DistanceSquared(&sp44, &PLAYER->actor.world.pos), 255, 20, 150);
+                sp3A = Quake_Add(globalCtx->cameraPtrs[globalCtx->activeCamera], 3);
+                Quake_SetSpeed(sp3A, 0x4E20);
+                Quake_SetQuakeValues(sp3A, 7, 0, 0, 0);
+                Quake_SetCountdown(sp3A, 12);
+                if (this->unk_179 <= 0) {
+                    func_8019F128(NA_SE_EV_STONEDOOR_STOP);
+                    this->unk_179 = 40;
+                }
+                break;
+            }
+        }
+    }
+}
+
+void func_80A9C18C(BgHakuginPost* this, GlobalContext* globalCtx) {
+    s32 pad;
+    Player* player = PLAYER;
+    Camera* camera = ACTIVE_CAM;
```

Comment:
```suggestion
    Camera* activeCam = ACTIVE_CAM;
```
This is something new I started to pick up during my docs of OoTs camera, but I'm starting to specify which camera is used, either `mainCam`, `activeCam`, or `subCam`